ui/articles-aem-renderer/src/androidTest/java/org/cru/godtools/articles/aem/db/ArticleDBTest.java

```diff
@@ -0,0 +1,145 @@
+                        .parse("Fri Jun 08 2018 18:55:00 GMT+0000").getTime();
+                article.mDateUpdated = new SimpleDateFormat("E MMM dd yyyy HH:mm:ss zz")
+                        .parse("Sat May 19 2018 00:23:39 GMT+0000").getTime();
+            } catch (ParseException e) {
+                Timber.e(e);
+                article.mDateCreated = Calendar.getInstance().getTimeInMillis();
+            }
+            Calendar calendar = Calendar.getInstance();
+            calendar.getTime();
+            calendar.setTimeInMillis(article.mDateCreated);
+            Date date = new Date(article.mDateCreated);
+            date.getYear();
+            date.getDate();
+            date.getHours();
+            date.getTimezoneOffset();
+            article.mTitle = " The title = " + i;
+            mArticleDao.insertArticle(article);
+
+        }
+
+    }
+
+
+    private void associateToArticles(List<Article> savedArticles) {
+        for (Article article : savedArticles != null ? savedArticles : new ArrayList<Article>()) {
+            // Create an attachment for each
+            Attachment attachment = new Attachment();
+            attachment.mArticleKey = article.mId;
+            attachment.mAttachmentUrl = "https://i.pinimg.com/originals/b9/9e/a0/b99ea0abc5e6bf5f5c67c1797791cc82.jpg";
+            mAttachmentDao.insertAttachment(attachment);
+
+            // Create Manifest Association
+            ManifestAssociation association = new ManifestAssociation();
+            association.mArticleId = article.mId;
+            association.mManifestId = String.valueOf(article.mId % 3);  // just need to only have three versions
+            association.mManifestName = String.format("Manifest # %s", association.mManifestId);
+            mAssociationDao.insertAssociation(association);
+        }
+    }
+
+    private void getAllSavedArticles() {
+        mArticleDao.getAllArticles().observe(this, articles -> {
+            associateToArticles(articles);
+
+            //verify that we are getting bat the Articles
+            Boolean allAttachmentsSaved = false;
+
+            for (Article article : articles) {
+                if (mAttachmentDao.getAttachmentsByArticle(article.mId) == null) {
+                    allAttachmentsSaved = false;
+                    break;
+                }
+
+                allAttachmentsSaved = true;
+            }
+
+            // verify that all of the Articles are retr
+
+            assert (Objects.requireNonNull(articles).size() > 0 &&
+                    allAttachmentsSaved &&
+                    Objects.requireNonNull(mAssociationDao
+                            .getArticlesByManifestID("0").getValue()).size() > 0 &&
+                    Objects.requireNonNull(mAssociationDao
+                            .getAssociationByManifestID("1").getValue()).size() > 0);
+        });
+    }
+
+
+    @Override
+    public Lifecycle getLifecycle() {
```

Comment:
why did you need to instrument this with `LifecycleObserver`?

Reply:
I was having trouble testing the Live Dat in the Test I wrote a task with Immutable data and it worked fine.